src/screens/WalletDetailsScreen.tsx

```diff
@@ -42,7 +45,7 @@ export class WalletDetailsScreen extends React.PureComponent<Props> {
               containerStyle={styles.showWalletXPUBContainer}
             />
             <FlatButton
-              onPress={this.deleteWallet}
+              onPress={this.deleteWallet.bind(this, wallet)}
```

Comment:
If `wallet` comes from props, you do not need to bind function in the render but just access 
```
const wallet = this.props.navigation.getParam('wallet');
```
in `deleteWallet = (wallet: Wallet) =>`

Remember, that every time you bind a function or create a arrow function in `render`, a small kitten dies